plugins/dynamix/scripts/newperms

```diff
@@ -17,34 +17,34 @@
 #  ugo+X   Set the 'x' bit for directories in user, group, and other
 
 function process {
-  echo "$1"
-  if [ -d "$1" ] ; then
-    owner=nobody
-    group=users
-    if [ $# -ge 2 ]; then
-      owner=$2
+  list="$1"
+  [[ $# -ge 2 ]] && owner=$2 || owner=nobody
+  [[ $# -ge 3 ]] && group=$3 || group=users
+  ifs=$IFS; IFS='|'
+  for dir in $list; do
+    echo "$dir"
+    if [[ -d $dir ]]; then
+      echo processing "$dir"
+      echo ... chmod -R u-x,go-rwx,go+u,ugo+X "$dir"
+      chmod -R u-x,go-rwx,go+u,ugo+X "$dir"
+      echo ... chown -R $owner:$group "$dir"
+      chown -R $owner:$group "$dir"
+      echo ... sync
+      sync
     fi
-    if [ $# -ge 3 ]; then
-      group=$3
-    fi
-    echo processing "$1"
-    echo ... chmod -R u-x,go-rwx,go+u,ugo+X "$1"
-    chmod -R u-x,go-rwx,go+u,ugo+X "$1"
-    echo ... chown -R $owner:$group "$1"
-    chown -R $owner:$group "$1"
-    echo ... sync
-    sync
-  fi
+  done
+  IFS=$ifs
 }
 
-if [ -n "$1" ] ; then
+if [[ $# -ge 1 ]]; then
   process "$@"
 else
   process /mnt/cache
-  for disk in /mnt/disk* ; do
+  for disk in /mnt/disk[0-9]*; do
```

Comment:
Look like you also fixed a bug preventing `/mnt/disks` (folder used by Unassigned Devices plugin) from being affected by new permissions.  Nice work!